File: LeetCode Problems/1701 to 1800/1704. Determine if String Halves Are Alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s) {
        string a = s.substr(0, s.size() / 2);
        string b = s.substr(s.size() / 2, s.size());
        int cnt = 0;
        for (auto ele : a){
            char ch = tolower(ele);
            if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u')
                cnt++;
        }
        for (auto ele : b){
            char ch = tolower(ele);
            if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u')
                cnt--;
        }
        if (cnt == 0)
            return true;
        else
            return false;
    }
};

File: LeetCode Problems/1701 to 1800/1704. Determine if String Halves Are Alike/1704. Determine if String Halves Are Alike.py
class Solution:
    def halvesAreAlike(self, s, vwls=set(list("aeiouAEIOU"))):    
        return (sum(c in vwls for c in s[:len(s)//2]) == sum(c in vwls for c in s[len(s)//2:]))
